rest/blip_api_crud_test.go

```diff
@@ -3151,3 +3151,92 @@ func TestOnDemandImportBlipFailure(t *testing.T) {
 		}
 	})
 }
+
+// TestBlipDatabaseClose verifies that the client connection is closed when the database is closed.
+// Starts a continuous pull replication then updates the db to trigger a close.
+func TestBlipDatabaseClose(t *testing.T) {
+
+	base.SetUpTestLogging(t, base.LevelInfo, base.KeyHTTP, base.KeySync, base.KeySyncMsg, base.KeyChanges, base.KeyCache)
+
+	bt, err := NewBlipTester(t)
```

Comment:
I am not convinced this test is actually testing what we want to be testing - I think to test this there are few things we want to change:

1. Use persistent config, or parameterize this to use either, because the persistent and non persistent config use different pathways a bit:
```
rt := NewRestTesterPersistentConfig()
defer rt.Close()
bt := NewBlipTesterFromSpecWithRT(rt.TB(), nil, rt)
```
2. Potentially parameterize how the database is shut down. The changing db config and taking database config both use `_removeDatabase` might be different.
   - `RemoveDatabase` (but it might be more clear to use `DELETE /db/`, even though this is a pass through call.
   - `rt.TakeDbOffline()` (this is a POST `/db/_offline`)
   - `rt.UpsertDbConfig()` (this will use `ReloadDatabaseWithConfig`)
3. Doing a pull replication (continuous/non continuous) and a push replication. We want to test an idle connection hangup, or a non idle connection hangup. It might make sense to be continually sending changes. I think there's a reasonable chance that a pull replication is already cancelled because `changeListener.Stop` will close `changeListener.terminator` and then the changes feeds are woken up with `changeListener.tapNotifier.Broadcast` which will return `WaiterCheckTerminated`. Does this new test pass without the go blip changes?
4. If deciding to test the `TakeDbOffline` pathway, we should make sure that taking it online again make it usable.

Reply:
I am convinced this test is actually doing the right thing, since I was confused about https://github.com/couchbase/sync_gateway/pull/7157/files#r1800260666.